packages/authentication/modules/access/usePreAuth/index.ts

```diff
@@ -0,0 +1,60 @@
+import { ACCESS_COOKIE_NAME, REFRESH_COOKIE_NAME, TokenTypes } from '@baseapp-frontend/utils'
+
+import { useQuery, useQueryClient } from '@tanstack/react-query'
+import Cookies from 'js-cookie'
+
+import AuthApi from '../../../services/auth'
+import { USER_API_KEY } from '../../../services/user'
+import { useSimpleTokenUser } from '../../user'
+import { IUsePreAuth } from './types'
+import { isJWTResponse } from './utils'
+
+const usePreAuth = ({
+  token,
+  queryOptions = {},
+  tokenType = TokenTypes.jwt,
+  cookieName = ACCESS_COOKIE_NAME,
+  refreshCookieName = REFRESH_COOKIE_NAME,
+  ApiClass = AuthApi,
+}: IUsePreAuth) => {
+  const queryClient = useQueryClient()
+  const { refetch: refetchUser } = useSimpleTokenUser({ options: { enabled: false } })
+
+  const query = useQuery({
+    staleTime: Infinity, // makes cache never expire automatically
+    enabled: !!token,
```

Comment:
do we need that? Normally this endpoint doesn't require/allow a token. I think that would stop people who don't have any token set to run that query, no?